### DISCUSSION TITLE
Get notifications when a new version of electricitymap is deployed

### DISCUSSION THREAD

**Discussion by madsnedergaard** [electricitymaps/electricitymaps-contrib#3245] [Ideas] [answered]
## Goal

See if we can come up with a nice way that people can get notified on new releases. It should handle periods with lots of releases in a nice way.

## User story

As a contributor, I would like to know when a change I (or someone else) did to the codebase will appear on the actual map.

## Questions

How do you imagine this could be done in the best way? :)
I think automated emails on each deployment will be too much, but maybe our Slack could be used in a good way?

**Comment by FelixDQ** [comment]
Could we find a way to use the [release tags](https://github.com/tmrowco/electricitymap-contrib/releases) on Github? We could add a release every time we pull in the new changes

**Reply by madsnedergaard** [reply]
I think we could also make our CI write a Slack message - but since it takes some time before the parsers run, maybe we should not "notify" anyone before it has run? That complicates things though :)

**Reply by FelixDQ** [reply]
I think notifying immediately after we have pulled in the changes is fine. The changes should be visible on the map in under 1 hour after that unless the system is down. 

If we use Slack, we could just add a disclaimer that it takes up to an hour for parser changes to appear.

**Reply by FelixDQ** [reply]
I think that we can do the release tagging fairly easily with gh actions. I can try to make a quick prototype.

**Reply by FelixDQ** [reply]
This gh action will add a release tag to releases when we pull in changes:
https://github.com/FelixDQ/tag-submodule-on-release-action

Would this solve the problem you think?

**Reply by madsnedergaard** [reply]
I think that's a great way of knowing when it's released - next question then is where the information should go and if it should be for every single deployment or not. Often there's small capacity updates or a translation update, which is not super interesting... But it could also be difficult to determine what's "notification-worthy" if we only push it for certain things 🤔

**Reply by FelixDQ** [reply]
Yea, it might be difficult to determine if it is notification worthy 🤔

What if we:
* Disabled the current PR notifications in slack (I don't think that they provide a lot of value)
* Add the gh action I linked to add release tags every time we pull in the new changes. I could extend it to include a list of merged PRs in the description.
* Add a notification on Slack when we make new releases.

**Comment by madsnedergaard** [comment, marked as answer]
Hey @brandongalbraith, the work by @FelixDQ is now live and working, so whenever a new version is deployed we're creating a new release: https://github.com/tmrowco/electricitymap-contrib/releases

And you can watch the repo for new releases here:
<img width="342" alt="Screenshot 2021-08-12 at 15 37 59" src="https://user-images.githubusercontent.com/3296643/129207056-98fd2dde-6bbd-4d33-8404-6c4e3e4a204b.png">


Right now it's 10ish minutes delayed from release to deployment, but we hope to remove that delay eventually!

**Reply by brandongalbraith** [reply]
Thank you so much @madsnedergaard and @FelixDQ for the effort that was put into this. I received my first Release email yesterday morning right before this notification! Please let me know if there's anything I can do to compensate Tomorrow for the work that went into this.

![image](https://user-images.githubusercontent.com/350893/129398977-118b4825-4d1a-4608-ac85-274af5a91bde.png)

**Comment by clawfire** [comment]
There's also an RSS feed to subscribe to for releases 
https://github.com/electricitymap/electricitymap-contrib/releases.atom

In Slack your can `/feed add https://github.com/electricitymap/electricitymap-contrib/releases.atom` to follow it and get notified
